admin/api/src/main/java/org/codice/ddf/admin/api/config/sources/SourceConfiguration.java

```diff
@@ -0,0 +1,132 @@
+
+import static org.codice.ddf.admin.api.validation.ValidationUtils.validateHostName;
+import static org.codice.ddf.admin.api.validation.ValidationUtils.validatePort;
+import static org.codice.ddf.admin.api.validation.ValidationUtils.validateString;
+import static org.codice.ddf.admin.api.validation.ValidationUtils.validateUrl;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import org.codice.ddf.admin.api.config.Configuration;
+import org.codice.ddf.admin.api.config.ConfigurationType;
+import org.codice.ddf.admin.api.handler.ConfigurationMessage;
+import org.codice.ddf.admin.api.validation.ValidationUtils;
+
+import com.google.common.collect.ImmutableMap;
+
+public class SourceConfiguration extends Configuration {
+
+    public static final String CONFIGURATION_TYPE = "sources";
+
+    public static final String SOURCE_NAME = "sourceName";
+    public static final String SOURCE_HOSTNAME = "sourceHostName";
+    public static final String PORT = "sourcePort";
+    public static final String USERNAME = "sourceUserName";
+    public static final String PASSWORD = "sourceUserPassword";
+    public static final String ENDPOINT_URL = "endpointUrl";
+
+    private String sourceName;
+    private String sourceHostName;
+    private int sourcePort;
+    private String sourceUserName;
+    private String sourceUserPassword;
+    private String endpointUrl;
+
+    public SourceConfiguration() {
+
+    }
+
+    public SourceConfiguration(SourceConfiguration sourceConfiguration) {
+        super(sourceConfiguration);
+        this.sourceName = sourceConfiguration.sourceName;
+        this.sourceHostName = sourceConfiguration.sourceHostName;
+        this.sourcePort = sourceConfiguration.sourcePort;
+        this.sourceUserName = sourceConfiguration.sourceUserName;
+        this.sourceUserPassword = sourceConfiguration.sourceUserPassword;
+        this.endpointUrl = sourceConfiguration.endpointUrl;
+    }
+
+    public static <T extends SourceConfiguration> Map<String, Function<T, List<ConfigurationMessage>>> getBaseFieldValidationMap() {
+        return new ImmutableMap.Builder<String, Function<T, List<ConfigurationMessage>>>()
+                .put(SOURCE_NAME, config -> validateString(config.sourceName(), SOURCE_NAME))
+                .put(SOURCE_HOSTNAME, config -> validateHostName(config.sourceHostName(),
+                        SOURCE_HOSTNAME))
+                .put(PORT, config -> validatePort(config.sourcePort(), PORT))
+                .put(USERNAME, config -> validateString(config.sourceUserName(), USERNAME))
+                .put(PASSWORD, config -> validateString(config.sourceUserPassword(), PASSWORD))
+                .put(ENDPOINT_URL, config -> validateUrl(config.endpointUrl(), ENDPOINT_URL))
+                .put(SERVICE_PID, config -> validateString(config.servicePid(), SERVICE_PID))
+                .build();
+    }
+
+    public List<ConfigurationMessage> validate(List<String> fields) {
```

Comment:
miss read your comment. This validation is still being used by the sources wizard before the source type is known.